src/Magento/FunctionalTestingFramework/Test/Util/ActionObjectExtractor.php

```diff
@@ -167,6 +169,42 @@ private function processActionGroupArgs($actionType, $actionAttributeData)
         return $actionAttributeArgData;
     }
 
+    /**
+     * Takes the helper arguments as an array that can be passed to PHP class
+     * defined in the action group xml.
+     *
+     * @param string $actionType
+     * @param array  $actionAttributeData
+     * @return array
+     * @throws TestFrameworkException
+     */
+    private function processHelperArgs($actionType, $actionAttributeData)
+    {
+        $reservedHelperVariableNames = ['class', 'method'];
+        if ($actionType !== self::HELPER_TAG) {
+            return $actionAttributeData;
+        }
+
+        $actionAttributeArgData = [];
+        foreach ($actionAttributeData as $attributeDataKey => $attributeDataValues) {
+            if (isset($attributeDataValues['nodeName']) && $attributeDataValues['nodeName'] == 'argument') {
+                if (isset($attributeDataValues['name'])
+                    && in_array($attributeDataValues['name'], $reservedHelperVariableNames)) {
+                    $message = 'Helper argument names ' . implode(',', $reservedHelperVariableNames);
+                    $message .= ' are reserved and can not be used.';
+                    throw new TestFrameworkException(
+                        $message
+                    );
```

Comment:
Do we need to keep track of all custom methods and throw error when there is an existing one?

Reply:
Didn't get your question 

This exception will throw an error when someone will try to use argument with name "class" or "method". Also they will be notified in PHPStorm, because XSD schema includes this verification as well.

Reply:
Yes. I know the reason for the exception.

My original question was what if two extensions both define a custom helper with method "goTo", now I realized they probably have different namespace. Never mind.